cpu/lpc2387/ldscripts/lpc2387.ld

```diff
@@ -198,12 +129,56 @@ SECTIONS
         PROVIDE(__stack_end = .);
     } > ram
 
+    .relocate :
+    {
+        . = ALIGN(4);
+        _srelocate = .;
+        *(.ramfunc .ramfunc.*);
+        *(.data .data.*);
+        KEEP (*(.openocd .openocd.*))
+        . = ALIGN(4);
+        _erelocate = .;
+    } > ram AT> rom
+
+    /* .bss section which is used for uninitialized data */
+    .bss (NOLOAD) :
+    {
+        . = ALIGN(4);
+        _sbss = . ;
+        _szero = .;
+        *(.bss .bss.*)
+        *(COMMON)
+        . = ALIGN(4);
+        _ebss = . ;
+        _ezero = .;
+    } > ram
+
+    /*
+     * collect all uninitialized sections that go into RAM
+     */
+    .noinit (NOLOAD) :
+    {
+        __noinit_start = .;
+        PROVIDE(__fiq_handler = .);
+        *(.fiq)
+        *(.noinit)
+        . = ALIGN(4);
+        __noinit_end = .;
+    }  > ram
+
+    /* heap section */
+    . = ALIGN(4);
+    _sheap = . ;
+    _eheap = ORIGIN(ram) + LENGTH(ram);
+
+    /* Populate information about ram size */
+    _sram = ORIGIN(ram);
+    _eram = ORIGIN(ram) + LENGTH(ram);
+
     __heap2_size = LENGTH(ram_ethernet);
     .heap2 (NOLOAD) :
     {
         PROVIDE(__heap2_start = . );
-        . = . + __heap2_size;
-        PROVIDE(__heap2_max = .);       /* _heap shall always be < _heap_max */
```

Comment:
No need to 'fill' those sections just to provide `__heap2_max`.
Having `__heap2_size` is enough.

This removes the bogus added 32k from the `size` of the elf file.